platform-sdk/docs/core/tipset-algorithm.md

```diff
@@ -0,0 +1,260 @@
+
+The "tipset algorithm" is a strategy for creating events that provides several key properties:
+
+- partition resiliency, provided by stopping the creation of events when a quorum is lost
+- improved consensus latency, provided by choosing hashgraph topologies that cause consensus to quickly advance
+- automatic event creation throttles that limit event creation rates faster than what helps consensus
+
+# Terms And Primitive Operations
+
+## Tipset
+
+A tipset is similar to a vector clock. It is an array of integer values.
+
+For a particular network, there will be one entry in a tipset for each node in the address book.
+The value of each entry corresponds to an event generation. The index in the tipset corresponds to
+each node's index in the address book.
+
+
+
+```
+Example
+
+For an address book with 4 nodes: A, B, C, and D
+
+[1, 3, 5, 2]
+       ^
+       |
+       Node C has a generation of 5 in this tipset
+```
+
+## Tipset of an Event
+
+Each event in the hashgraph has a tipset.
+
+To calculate the tipset of an event, iterate over all ancestors of the event (in this context,
+an event is considered to be an ancestor to itself). From all of these ancestors, sort
+by event creator. For each event creator, choose the event with the highest generation. The entry for that node
+in the tipset is equal to the generation of that event. If there are no ancestors from a particular event
+creator, the entry for that node in the tipset is -1.
+
+```
+Example
+
+Time starts at the bottom and moves forward as you go up the graph. Numbers for each event are generations.
+
+ |         (4)         |          |
+ |         /|          |          |
+ |        / |          |          |
+ |       /  |          |          |
+ |      /   |          |          |
+ |     /    |          |          |
+ |    /     |          |          |
+ |   /      |          |          |
+ |  /       |          |          |
+ | /        |          |          |
+(3)         |          |          |
+ |\         |          |          |
+ | \        |          |          |
+ |  \       |          |          |
+ |   \      |          |          |
+ |    \     |          |          |
+ |     \    |          |          |
+ |      \   |          |          |
+ |       \  |          |          |
+ |        \ |          |          |
+ |         \|          |          |
+ |         (2)         |         (2)
+ |          |\         |         /|
+ |          | \        |        / |
+ |          |  \       |       /  |
+ |          |   \      |      /   |
+ |          |    \     |     /    |
+ |          |     \    |    /     |
+ |          |      \   |   /      |
+ |          |       \  |  /       |
+ |          |        \ | /        |
+ |          |         \|/         |
+(1)        (1)        (1)         |
+ |          |          |          |
+ |          |          |          |
+ A          B          C          D
+
+ The tipset of B's most recent event is [3, 4, 1, -1]
```

Comment:
> Numbers for each event are generations.

This is mentioned above. This document assumes the reader is familiar with a number of hashgraph terms and concepts, and isn't designed to be easy to digest without that knowledge.

Instead of adding this sort of introductory material to this document, I'd prefer we spend the time to document the system as a whole.

That being said, before we get around to generating this documentation, I'm more than happy to hop on a zoom call to go over these core concepts.